### PR TITLE
IGNITE-20153 Prepare existing tests for the distributed zone to switch to the catalog

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-20153

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java [hunk 1]
```diff
@@ -161,4 +172,38 @@ void startDistributionZoneManager() {
...
+                dataNodesAutoAdjustScaleUp,
+                dataNodesAutoAdjustScaleDown,
+                filter
+        );
+    }
+
+    protected void dropZone(String zoneName) {
+        DistributionZonesTestUtil.dropZone(distributionZoneManager, zoneName);
+    }
+}
```
Comment (sanpwc) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java:209]: Empty line is missing.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java [hunk 2]
```diff
@@ -61,7 +62,15 @@
  * Base class for {@link DistributionZoneManager} unit tests.
  */
 @ExtendWith(ConfigurationExtension.class)
-public class BaseDistributionZoneManagerTest extends BaseIgniteAbstractTest {
+public abstract class BaseDistributionZoneManagerTest extends BaseIgniteAbstractTest {
+    protected static final String ZONE_NAME = "zone1";
+
+    protected static final int ZONE_ID = 1;
+
+    protected static final long TIMEOUT_MILLIS = 10_000L;
```
Comment (sanpwc) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java:70]: Please rename both TIMEOUT_MILLIS and TIMER_SECONDS it's not clear what it means.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java [hunk 3]
```diff
@@ -169,18 +172,20 @@ public void setUp() throws Exception {
 
     @AfterEach
     public void tearDown() throws Exception {
-        distributionZoneManager.stop();
-        metaStorageManager.stop();
-        clusterCfgMgr.stop();
-        vaultMgr.stop();
-        generator.close();
+        IgniteUtils.closeAll(
+                distributionZoneManager == null ? null : distributionZoneManager::stop,
```
Comment (sanpwc) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java:176]: Why do we need null check here? Do we expect any of closeables to be null in happy case?

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java [hunk 4]
```diff
@@ -113,11 +113,40 @@ public static CompletableFuture<Integer> createZone(
...
+     *
+     * @param distributionZoneManager Distributed zone manager.
+     * @param zoneName Zone name.
+     * @param dataNodesAutoAdjustScaleUp Timeout in seconds between node added topology event itself and data nodes switch,
+     *         {@code null} if not set.
+     * @param dataNodesAutoAdjustScaleDown Timeout in seconds between node left topology event itself and data nodes switch,
+     *         {@code null} if not set.
+     * @param filter Nodes filter, {@code null} if not set.
+     */
+    public static void createZone(
```
Comment (sanpwc) [inline modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java:135]: It's confusing to have both async and sync createZone methods with same createZone name. Do we really need async ones in tests?

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerScaleUpTest.java [hunk 5]
```diff
@@ -102,44 +99,32 @@ void testDataNodesPropagationAfterScaleUpTriggered() throws Exception {
...
-                        .dataNodesAutoAdjustScaleUp(IMMEDIATE_TIMER_VALUE)
-                        .dataNodesAutoAdjustScaleDown(INFINITE_TIMER_VALUE)
-                        .build()
-        ).get();
+        alterZone(DEFAULT_ZONE_NAME, IMMEDIATE_TIMER_VALUE, INFINITE_TIMER_VALUE, null);
 
-        distributionZoneManager.createZone(
-                new DistributionZoneConfigurationParameters.Builder(ZONE_1_NAME).dataNodesAutoAdjustScaleUp(IMMEDIATE_TIMER_VALUE).build()
-        ).get();
+        createZone(ZONE_NAME, IMMEDIATE_TIMER_VALUE, null, null);
```
Comment (sanpwc) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerScaleUpTest.java:122]: Why do you think that such shortcut worth writing it?  Original option is much more readable.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java [hunk 6]
```diff
@@ -111,498 +98,17 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void testCreateZoneWithAutoAdjust() throws Exception {
```
Comment (sanpwc) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java:114]: Why did you remove this and similar tests?

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java:209]: fix it

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItZoneDdlTest.java [hunk 7]
```diff
@@ -17,90 +17,116 @@
...
-    private static String randomZoneName() {
-        return "test_zone" + System.currentTimeMillis();
+    private static void tryToAlterZone(String zoneName, int dataNodesAutoAdjust, boolean failIfNotExists) {
+        sql(String.format(
+                "ALTER ZONE %s SET DATA_NODES_AUTO_ADJUST=%s",
+                failIfNotExists ? zoneName : "IF EXISTS " + zoneName, dataNodesAutoAdjust
+        ));
     }
-}
+}
```
Comment (sanpwc) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItZoneDdlTest.java:132]: Empty line.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java (hunk 3 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java:176]: Since the test can fail at any step, including initialization, checking for `null` looks like the right approach.
Checking/expecting only a happy case doesn't seem right to me.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java (hunk 2 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java:70]: What names do you suggest?
TIMEOUT_MILLIS - use for IgniteTestUtils#waitForCondition
TIMER_SECONDS - use for scale up and down

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java (hunk 4 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java:135]: Async method will leave after switch to the catalog.
I propose to leave it as it is, I can change it to synchronous, wdyt?

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java (hunk 3 above)
Comment (sanpwc) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java:176]: closeAll() internally will handle NPE in order to close all the resources, won't it?

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java (hunk 4 above)
Comment (sanpwc) [reply modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java:134]: Not sure that I get it. Do we use the async nature of given method somewhere? If not please change it to synchronous. The main point here is to have consistent contract.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java (hunk 3 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java:176]: You are right, but there is a nuance, it is necessary for the class to implement `AutoCloseable`, none of them do this.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java (hunk 2 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/BaseDistributionZoneManagerTest.java:70]: Did it

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java (hunk 6 above)
Comment (sanpwc) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java:114]: There are two major points here:
1. First of all, DZM has it's own API that should be tested.
2. Highly likely some zone specific validation logic will be presented in DZM itself, so it should be tested though corresponding API.

Comment (tkalkirill) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java:114]: Discussed personally - delete them.

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java (hunk 4 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java:134]: fix it

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java (hunk 3 above)
Comment (sanpwc) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java:176]: Yep, you are right.

**Review by sanpwc — APPROVED** [review]
